plugins/net_plugin/net_plugin.cpp

```diff
@@ -537,6 +542,28 @@ namespace eosio {
       void start_expire_timer();
       void start_monitors();
 
+      void wait_if_paused() {
+         controller& cc = chain_plug->chain();
+         if (!cc.is_writing_snapshot())
+            return;
+
+         std::unique_lock l(pause_mtx);
+         paused = true;
+         while (!pause_cv.wait_for(l, std::chrono::milliseconds(10), [&]{ return !paused; } )) {
+            if (!cc.is_writing_snapshot()) {
+               paused = false;
+            }
+         }
+      }
```

Comment:
It occurred to me that this could just be:
```cpp
      void wait_if_paused() {
         controller& cc = chain_plug->chain();
         if (cc.is_writing_snapshot()) {
            std::this_thread::sleep_for( 10ms );
         }
      }
```
And not use a `condition_variable` at all. Let me know if you prefer the simpler implementation.

Reply:
Yes I would vote for the simple sleep loop in this case since it's already doing something effectively the same via `wait_for()` anyways. If we're going to add any complexity beyond a simple sleep loop it might be best as a signal from controller when a snapshot is complete so we don't need to loop (not suggesting we do that).